pkg/services/alerting/notifiers/line.go

```diff
@@ -70,27 +77,67 @@ func (this *LineNotifier) createAlert(evalContext *alerting.EvalContext) error {
 		return err
 	}
 
-	form := url.Values{}
-	body := fmt.Sprintf("%s - %s\n%s", evalContext.Rule.Name, ruleUrl, evalContext.Rule.Message)
-	form.Add("message", body)
+	message := fmt.Sprintf("%s - %s\n%s", evalContext.Rule.Name, ruleUrl, evalContext.Rule.Message)
+	var body string
+	header := map[string]string{
+		"Content-Type":  "application/x-www-form-urlencoded;charset=UTF-8",
+		"Authorization": fmt.Sprintf("Bearer %s", this.Token),
+	}
 
+	// send message only if image is not included
+	if !this.UploadImage {
+		form := url.Values{}
+		form.Add("message", message)
+		body = form.Encode()
+	} else
+	// add image url if external image uploader is set
 	if evalContext.ImagePublicUrl != "" {
+		form := url.Values{}
+		form.Add("message", message)
 		form.Add("imageThumbnail", evalContext.ImagePublicUrl)
 		form.Add("imageFullsize", evalContext.ImagePublicUrl)
+		body = form.Encode()
+	} else {
+		// upload image to line server
+		var file string
+		if evalContext.ImageOnDiskPath == "" {
+			file = filepath.Join(setting.HomePath, "public/img/mixed_styles.png")
```

Comment:
I don't think use default image is a good idea since it provides no information, just break the condition and use default body may be better

Reply:
Agree.